### ISSUE TITLE
Explore different neural network models

### ISSUE THREAD

**Issue by bamos** [cmusatyalab/openface#78]
The original `nn4` model was trained with a lot of data that we currently don't have,
so naturally a different neural network architecture is probably best-suited
for the data we're currently training on.
This idea is partially validated by the manually-written `nn4.small` models
performing better than the `nn4` model.
This issue explores automatic ways of creating smaller networks
by using random hyper-parameter choices.

**Comment by jmorra** [comment]
As a suggestion have the authors of this project considered something along the lines of [this](https://github.com/hyperopt/hyperopt) for hyper-parameter optimization?

**Comment by ogrisel** [comment]
You might also be interested in a hosted service version of hyperopt with both Python and Lua client SDKs:

http://oscar.sensout.com/

**Comment by Martyn10** [comment]
The "Center Loss" function proposed here looks interesting. Seems like it would be much easier to train compared to triplet loss:
http://ydwen.github.io/papers/WenECCV16.pdf

On his site he said he plans to release code in the future:
http://ydwen.github.io/

**Comment by bamos** [comment]
Thanks for the reference @Martyn10! We'll further look into using the center loss.
